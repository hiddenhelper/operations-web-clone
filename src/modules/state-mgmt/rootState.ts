import { History } from 'history';

import { IState as IAuthState } from './auth/state';
import { IState as IGeneralState } from './general/state';
import { IState as IClientState } from './client/state';
import { IState as IProjectState } from './project/state';
import { IState as IUserState } from './user/state';
import { IState as IAccessControlSystemState } from './access-control-system/state';
import { IState as IBadgePrintingSystemState } from './badge-printer-system/state';
import { IState as IWorkerState } from './worker/state';
import { IState as ICertificationState } from './certification/state';
import { IState as IFileState } from './file/state';
import { IState as IBadgeState } from './badge/state';
import { IState as IStatisticsState } from './statistics/state';
import { IState as ITrainingState } from './training/state';
import { IState as IPaymentState } from './payment/state';
import { IState as IInvoiceState } from './invoice/state';
import { ApiService } from '../services/ApiService';
import { AuthService } from '../services/AuthService';
import { PrinterService } from '../services/PrinterService';
import { FileService } from '../services/FileService';
import { Logger } from '../services/Logger';
import { IProcoreState } from './procore/state';
import { IState as IProjectNewState } from './project-new/state';

export interface IAction {
  type: string;
  payload: any;
}

export interface IRootState {
  auth: IAuthState;
  user: IUserState;
  client: IClientState;
  project: IProjectState;
  accessControlSystem: IAccessControlSystemState;
  badgePrinterSystem: IBadgePrintingSystemState;
  worker: IWorkerState;
  certification: ICertificationState;
  training: ITrainingState;
  invoice: IInvoiceState;
  file: IFileState;
  badge: IBadgeState;
  statistics: IStatisticsState;
  payment: IPaymentState;
  general: IGeneralState;
  router?: { location: { pathname: string; search: string; hard: string; key: string }; action: string };
  procore: IProcoreState;
  projectNew: IProjectNewState;
}

export interface IEpicDependencies {
  history: History;
  apiService: ApiService;
  authService: AuthService;
  printerService: PrinterService;
  fileService: FileService;
  logger: typeof Logger;
}
