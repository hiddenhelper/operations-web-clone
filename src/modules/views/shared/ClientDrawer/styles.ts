import { createStyles, makeStyles } from '@material-ui/core/styles';
import { STYLE } from '../../../../constants';
import { toREM } from '../../../../utils/generalUtils';

export const useStyles = makeStyles(theme =>
  createStyles({
    title: {
      wordBreak: 'break-all',
      color: STYLE.COLOR.SECONDARY,
    },
    cancelButtonWrapper: {
      display: 'flex',
      justifyContent: 'flex-start',
      alignItems: 'center',
    },
    closeButton: {
      width: `${toREM(80)}`,
      height: `${toREM(24)}`,

      '&:hover': {
        backgroundColor: 'initial',
      },
    },
    buttonsWrapper: {
      display: 'flex',
      justifyContent: 'flex-end',
    },
    primaryButton: {
      marginLeft: `${toREM(19)}`,
    },
  })
);
