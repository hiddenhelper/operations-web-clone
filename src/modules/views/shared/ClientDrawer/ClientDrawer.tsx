import React, { memo, useCallback, useEffect, useMemo, useState } from 'react';

import Divider from '@material-ui/core/Divider';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import PersonIcon from '@material-ui/icons/Person';
import EmailIcon from '@material-ui/icons/Email';

import ControlledButton from '../FormHandler/ControlledButton';
import Drawer from '../ResourceManagement/Drawer';
import Modal from '../Modal';
import DeleteModal from '../Modal/components/Delete';
import Address from '../Address';

import { LocationIcon } from '../../../../constants';
import { ClientModel, GeneralModel } from '../../../models';
import { useStyles as modalStyles } from '../Modal/style';
import { useStyles as buttonStyles } from '../FormHandler/ControlledButton/styles';
import { getTradesString } from '../../../../utils/tradeUtils';
import { useStyles as drawerStyles } from '../ResourceManagement/Drawer/styles';
import { listGlobalStyles } from '../../../../assets/styles';
import { formatPhoneNumber, getConditionalDefaultValue, getDefaultValue } from '../../../../utils/generalUtils';

export interface IClientDrawerProps {
  client: ClientModel.IClient;
  mwbeList: GeneralModel.INamedEntity[];
  clientListElement: React.ReactNode;
  isOpen: boolean;
  isLoading: boolean;
  showTaxExempt?: boolean;
  buttonText?: string;
  buttonTestId?: string;
  showPrimaryButton?: boolean;
  showSecondaryButton?: boolean;
  deleteLoading?: GeneralModel.ILoadingStatus;
  onClose: () => void;
  onDelete?: (id: string) => void;
  handleButtonClick?: () => void;
}

const ClientDrawer = ({
  isOpen,
  isLoading,
  showTaxExempt,
  deleteLoading,
  client,
  mwbeList,
  clientListElement,
  buttonText,
  buttonTestId,
  showPrimaryButton,
  showSecondaryButton,
  onClose,
  onDelete,
  handleButtonClick,
}: IClientDrawerProps) => {
  const clientListHeight = (clientListElement as any)?.offsetHeight;
  const drawerClasses = drawerStyles();
  const listClasses = listGlobalStyles();
  const modalClasses = modalStyles();
  const buttonClasses = buttonStyles();
  const [showDeleteModal, setShowDeleteModal] = useState(false);
  const onDeleteOpen = useCallback(() => {
    setShowDeleteModal(true);
  }, []);
  const currentMwbe = useMemo(() => mwbeList && mwbeList.find(option => option.id === client.mwbeTypeId), [mwbeList, client]);
  const tradeList = useMemo(() => getTradesString(client.trades, client.otherTrade), [client.trades, client.otherTrade]);

  const closeModal = useCallback(/* istanbul ignore next */ () => setShowDeleteModal(false), [setShowDeleteModal]);

  const handleDeleteAccepted = useCallback(() => {
    onDelete(client.id);
  }, [onDelete, client.id]);

  useEffect(() => {
    if (deleteLoading && !deleteLoading.isLoading) {
      setShowDeleteModal(false);
    }
  }, [deleteLoading]);

  return (
    <>
      <Drawer
        title="Client Information"
        dataTestId="client-drawer-detail"
        height={clientListHeight}
        isOpen={isOpen}
        isLoading={isLoading}
        onClose={onClose}
        render={() => (
          <div className={listClasses.listDetail}>
            <div className={listClasses.company}>
              <div className={drawerClasses.drawerText}>
                <Typography className={`${listClasses.listAccent} ${listClasses.entityTitle}`}>{client.name}</Typography>
                <div className={drawerClasses.drawerMainText}>
                  <span>{tradeList}</span>
                  <span>{client.taxpayerIdentificationNumber ? client.taxpayerIdentificationNumber : '-'}</span>
                  <span>{currentMwbe?.name}</span>
                  {showTaxExempt && client.isTaxExempt && <span>Tax Exempt</span>}
                </div>
              </div>

              <div className={drawerClasses.drawerSection}>
                <LocationIcon />
                <div className={`${drawerClasses.drawerText} ${drawerClasses.drawerTextSeparation}`}>
                  {getConditionalDefaultValue(client.billingAddress, <Address address={client.billingAddress} />)}
                </div>
              </div>
              <div className={drawerClasses.drawerSection}>
                <EmailIcon />
                <div className={`${drawerClasses.drawerText} ${drawerClasses.drawerTextSeparation}`}>
                  {getConditionalDefaultValue(client.mailingAddress, <Address address={client.mailingAddress} />)}
                </div>
              </div>
              <div className={drawerClasses.drawerSection}>
                <PersonIcon />
                <div className={`${drawerClasses.drawerText} ${drawerClasses.drawerTextSeparation}`}>
                  <span className={listClasses.listAccent}>
                    {getConditionalDefaultValue(client.user, `${client?.user?.firstName} ${client?.user?.lastName}`)}
                  </span>
                  {<span>{getDefaultValue(client?.user?.email)}</span>}
                  {<span>{formatPhoneNumber(client?.user?.mobilePhoneNumber)}</span>}
                </div>
              </div>
            </div>
            {(showPrimaryButton || showSecondaryButton) && <Divider className={drawerClasses.drawerDivider} />}
            <div className={listClasses.ctaWrapper}>
              {showPrimaryButton && (
                <ControlledButton>
                  <Button
                    disableRipple={true}
                    className={`${buttonClasses.drawerCTA} ${buttonClasses.borderPrimaryButton}`}
                    variant="outlined"
                    onClick={handleButtonClick}
                    data-testid={buttonTestId}
                  >
                    {buttonText}
                  </Button>
                </ControlledButton>
              )}
              {showSecondaryButton && (
                <ControlledButton>
                  <Button
                    disableRipple={true}
                    className={`${buttonClasses.drawerCTA} ${buttonClasses.warningButton}`}
                    variant="outlined"
                    onClick={onDeleteOpen}
                    data-testid="deleteClientButton"
                  >
                    Delete
                  </Button>
                </ControlledButton>
              )}
            </div>
          </div>
        )}
      />
      <Modal
        show={showDeleteModal}
        styleClass={`${modalClasses.dialogContainer} ${modalClasses.deleteModal}`}
        render={() => <DeleteModal title={`Delete ${client.name}?`} onCancel={closeModal} onConfirm={handleDeleteAccepted} />}
      />
    </>
  );
};

export default memo(ClientDrawer);
