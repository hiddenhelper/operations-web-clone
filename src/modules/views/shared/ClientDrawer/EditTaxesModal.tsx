import React, { memo, useCallback, useState } from 'react';

import Grid from '@material-ui/core/Grid';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import Button from '@material-ui/core/Button';
import ButtonLoader from '../ButtonLoader';

import { useStyles } from './styles';
import ControlledSelect from '../FormHandler/ControlledSelect/ControlledSelect';
import { ClientModel } from '../../../models';

export interface IEditTaxesProps {
  taxCondition: boolean;
  loading: boolean;
  onCancel: () => void;
  onConfirm: (condition: boolean) => void;
}

const EditTaxes = ({ taxCondition, loading, onConfirm, onCancel }: IEditTaxesProps) => {
  const classes = useStyles();
  const [isTaxExempt, setIsTaxExempt] = useState(taxCondition);

  const handleTaxConditionChange = useCallback(
    event => {
      setIsTaxExempt(Boolean(event.target.value));
    },
    [setIsTaxExempt]
  );

  const handleConfirm = useCallback(() => {
    onConfirm(isTaxExempt);
  }, [onConfirm, isTaxExempt]);

  return (
    <>
      <DialogTitle id="alert-dialog-title" className={classes.title}>
        Edit Tax Condition
      </DialogTitle>
      <DialogContent>
        <ControlledSelect
          label=""
          name="isTaxExempt"
          value={Number(isTaxExempt)}
          options={ClientModel.taxExemptOptionList}
          onChange={handleTaxConditionChange}
          error={null}
          disabled={false}
          dataTestId="tax-condition-select"
        />
      </DialogContent>
      <DialogActions>
        <Grid container={true}>
          <Grid item={true} xs={3} className={classes.cancelButtonWrapper}>
            <Button className={classes.closeButton} data-testid="cancel-button" onClick={onCancel} color="primary">
              Close
            </Button>
          </Grid>
          <Grid item={true} xs={9} className={classes.buttonsWrapper}>
            <ButtonLoader
              text="Save"
              loadingText="Saving..."
              isLoading={loading}
              className={classes.primaryButton}
              data-testid="confirm-button"
              onClick={handleConfirm}
              color="primary"
              variant="contained"
            />
          </Grid>
        </Grid>
      </DialogActions>
    </>
  );
};

export default memo(EditTaxes);
