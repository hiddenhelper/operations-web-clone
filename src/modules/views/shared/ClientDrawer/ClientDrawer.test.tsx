import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import ClientDrawer, { IClientDrawerProps } from './ClientDrawer';
import { getClient_1 } from '../../../../test/entities';
import { createMockStore } from 'redux-test-utils';
import { getInitialState } from '../../../../test/rootState';
import { Provider } from 'react-redux';

describe('ClientDrawer', () => {
  let props: IClientDrawerProps;

  beforeEach(() => {
    props = {
      isOpen: true,
      isLoading: false,
      deleteLoading: undefined,
      client: getClient_1(),
      mwbeList: [],
      clientListElement: null,
      buttonText: 'Edit',
      buttonTestId: 'drawerClientButton',
      showPrimaryButton: true,
      showSecondaryButton: true,
      onClose: jest.fn(),
      onDelete: jest.fn(),
      handleButtonClick: jest.fn(),
    };
  });

  it('should render', () => {
    const { container } = render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <ClientDrawer {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render tax exempt', () => {
    props.showTaxExempt = true;
    props.client = { ...getClient_1(), isTaxExempt: true };
    const { container } = render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <ClientDrawer {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should show loading', () => {
    props.isLoading = true;
    props.deleteLoading = { isLoading: false, error: null, hasError: false };
    const wrapper = render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <ClientDrawer {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.container).toMatchSnapshot();
    expect(wrapper.getByText('Loading...'));
  });
});
