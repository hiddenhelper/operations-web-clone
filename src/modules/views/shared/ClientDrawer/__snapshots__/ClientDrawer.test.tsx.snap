// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientDrawer should render 1`] = `
<div>
  <div
    class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-1 open"
    data-testid="client-drawer-detail"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-2 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
    >
      <div
        class="makeStyles-drawerButtonWrapper-4"
      >
        <div
          class="makeStyles-drawerButton-5"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            data-testid="drawer-close-btn"
            style="outline: none;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <svg>
                close.svg
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="makeStyles-drawerHeader-3"
        >
          <p
            class="MuiTypography-root makeStyles-drawerTitle-6 MuiTypography-body1 MuiTypography-colorPrimary MuiTypography-alignLeft"
          >
            Client Information
          </p>
        </div>
      </div>
      <div
        class="makeStyles-listDetail-31"
      >
        <div
          class="makeStyles-company-32"
        >
          <div
            class="makeStyles-drawerText-8"
          >
            <p
              class="MuiTypography-root makeStyles-listAccent-33 makeStyles-entityTitle-30 MuiTypography-body1"
            >
              Robert C. Martin
            </p>
            <div
              class="makeStyles-drawerMainText-9"
            >
              <span>
                trade 1, trade 2
              </span>
              <span>
                LOZG-780211-7B9
              </span>
              <span />
            </div>
          </div>
          <div
            class="makeStyles-drawerSection-7"
          >
            <svg>
              location.svg
            </svg>
            <div
              class="makeStyles-drawerText-8 makeStyles-drawerTextSeparation-10"
            >
              <p
                class="MuiTypography-root makeStyles-cardFont-95  MuiTypography-body1"
              >
                
                <span
                  class=" "
                >
                  -
                </span>
              </p>
              <p
                class="MuiTypography-root makeStyles-cardFont-95  MuiTypography-body1"
              >
                
                
                
                
              </p>
            </div>
          </div>
          <div
            class="makeStyles-drawerSection-7"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
              />
            </svg>
            <div
              class="makeStyles-drawerText-8 makeStyles-drawerTextSeparation-10"
            >
              -
            </div>
          </div>
          <div
            class="makeStyles-drawerSection-7"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
            <div
              class="makeStyles-drawerText-8 makeStyles-drawerTextSeparation-10"
            >
              <span
                class="makeStyles-listAccent-33"
              >
                -
              </span>
              <span>
                -
              </span>
              <span>
                -
              </span>
            </div>
          </div>
        </div>
        <hr
          class="MuiDivider-root makeStyles-drawerDivider-11"
        />
        <div
          class="makeStyles-ctaWrapper-34"
        >
          <span
            class=""
            data-testid="controlled-button"
            tabindex="-1"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-drawerCTA-75 makeStyles-borderPrimaryButton-67"
              data-testid="drawerClientButton"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Edit
              </span>
            </button>
          </span>
          <span
            class=""
            data-testid="controlled-button"
            tabindex="-1"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-drawerCTA-75 makeStyles-warningButton-76"
              data-testid="deleteClientButton"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Delete
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClientDrawer should render tax exempt 1`] = `
<div>
  <div
    class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-107 open"
    data-testid="client-drawer-detail"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-108 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
    >
      <div
        class="makeStyles-drawerButtonWrapper-110"
      >
        <div
          class="makeStyles-drawerButton-111"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            data-testid="drawer-close-btn"
            style="outline: none;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <svg>
                close.svg
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="makeStyles-drawerHeader-109"
        >
          <p
            class="MuiTypography-root makeStyles-drawerTitle-112 MuiTypography-body1 MuiTypography-colorPrimary MuiTypography-alignLeft"
          >
            Client Information
          </p>
        </div>
      </div>
      <div
        class="makeStyles-listDetail-137"
      >
        <div
          class="makeStyles-company-138"
        >
          <div
            class="makeStyles-drawerText-114"
          >
            <p
              class="MuiTypography-root makeStyles-listAccent-139 makeStyles-entityTitle-136 MuiTypography-body1"
            >
              Robert C. Martin
            </p>
            <div
              class="makeStyles-drawerMainText-115"
            >
              <span>
                trade 1, trade 2
              </span>
              <span>
                LOZG-780211-7B9
              </span>
              <span />
              <span>
                Tax Exempt
              </span>
            </div>
          </div>
          <div
            class="makeStyles-drawerSection-113"
          >
            <svg>
              location.svg
            </svg>
            <div
              class="makeStyles-drawerText-114 makeStyles-drawerTextSeparation-116"
            >
              <p
                class="MuiTypography-root makeStyles-cardFont-201  MuiTypography-body1"
              >
                
                <span
                  class=" "
                >
                  -
                </span>
              </p>
              <p
                class="MuiTypography-root makeStyles-cardFont-201  MuiTypography-body1"
              >
                
                
                
                
              </p>
            </div>
          </div>
          <div
            class="makeStyles-drawerSection-113"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
              />
            </svg>
            <div
              class="makeStyles-drawerText-114 makeStyles-drawerTextSeparation-116"
            >
              -
            </div>
          </div>
          <div
            class="makeStyles-drawerSection-113"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
            <div
              class="makeStyles-drawerText-114 makeStyles-drawerTextSeparation-116"
            >
              <span
                class="makeStyles-listAccent-139"
              >
                -
              </span>
              <span>
                -
              </span>
              <span>
                -
              </span>
            </div>
          </div>
        </div>
        <hr
          class="MuiDivider-root makeStyles-drawerDivider-117"
        />
        <div
          class="makeStyles-ctaWrapper-140"
        >
          <span
            class=""
            data-testid="controlled-button"
            tabindex="-1"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-drawerCTA-181 makeStyles-borderPrimaryButton-173"
              data-testid="drawerClientButton"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Edit
              </span>
            </button>
          </span>
          <span
            class=""
            data-testid="controlled-button"
            tabindex="-1"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-drawerCTA-181 makeStyles-warningButton-182"
              data-testid="deleteClientButton"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Delete
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClientDrawer should show loading 1`] = `
<div>
  <div
    class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-213 open"
    data-testid="client-drawer-detail"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-214 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
    >
      <div
        class="makeStyles-drawerButtonWrapper-216"
      >
        <div
          class="makeStyles-drawerButton-217"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            data-testid="drawer-close-btn"
            style="outline: none;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <svg>
                close.svg
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="makeStyles-drawerHeader-215"
        >
          <p
            class="MuiTypography-root makeStyles-drawerTitle-218 MuiTypography-body1 MuiTypography-colorPrimary MuiTypography-alignLeft"
          >
            Client Information
          </p>
        </div>
      </div>
      <div>
        Loading...
      </div>
    </div>
  </div>
</div>
`;
